dbt/models/default/default.vw_pin_value.sql

```diff
@@ -1,190 +1,472 @@
             END
         ) AS board_tot,
         -- Board certified market values
         ARBITRARY(
             CASE
-                WHEN procname = 'BORVALUE' AND taxyr < '2020' THEN NULL
-                WHEN procname = 'BORVALUE' AND taxyr >= '2020' THEN valapr2
+                WHEN
+                    asmt.procname = 'BORVALUE' AND asmt.taxyr < '2020'
+                    THEN NULL
+                WHEN
+                    asmt.procname = 'BORVALUE' AND asmt.taxyr >= '2020'
+                    THEN asmt.valapr2
             END
         ) AS board_bldg_mv,
         ARBITRARY(
             CASE
-                WHEN procname = 'BORVALUE' AND taxyr < '2020' THEN NULL
-                WHEN procname = 'BORVALUE' AND taxyr >= '2020' THEN valapr1
+                WHEN
+                    asmt.procname = 'BORVALUE' AND asmt.taxyr < '2020'
+                    THEN NULL
+                WHEN
+                    asmt.procname = 'BORVALUE' AND asmt.taxyr >= '2020'
+                    THEN asmt.valapr1
             END
         ) AS board_land_mv,
         ARBITRARY(
             CASE
-                WHEN procname = 'BORVALUE' AND taxyr < '2020' THEN NULL
-                WHEN procname = 'BORVALUE' AND taxyr >= '2020' THEN valapr3
+                WHEN
+                    asmt.procname = 'BORVALUE' AND asmt.taxyr < '2020'
+                    THEN NULL
+                WHEN
+                    asmt.procname = 'BORVALUE' AND asmt.taxyr >= '2020'
+                    THEN asmt.valapr3
             END
         ) AS board_tot_mv
-    FROM {{ source('iasworld', 'asmt_all') }}
-    WHERE procname IN ('CCAOVALUE', 'CCAOFINAL', 'BORVALUE')
-        AND rolltype != 'RR'
-        AND deactivat IS NULL
-        AND valclass IS NULL
-    GROUP BY parid, taxyr
+    FROM {{ source('iasworld', 'asmt_all') }} AS asmt
+    LEFT JOIN stages
+        ON asmt.parid = stages.parid
+        AND asmt.taxyr = stages.taxyr
+    WHERE (
+        -- Check for two possible situations: Either the record is stamped
+        -- with a procname corresponding to a stage, in which case the record
+        -- corresponds to a final value for a stage, or the procname is null
+        -- and other fields indicate that it is a provisional value for an
+        -- upcoming stage
+        asmt.procname IN ('CCAOVALUE', 'CCAOFINAL', 'BORVALUE')
+        OR (
+            asmt.procname IS NULL
+            AND asmt.cur = 'Y'
+            AND (
+                -- If the PIN has no stages but its year is not the current
+                -- assessment year, it is likely a data error from a prior
+                -- year that we don't want to include in our results. In
+                -- contrast, if the PIN is in the current year but has no
+                -- stages, it is most likely a provisional value for a PIN
+                -- that has not mailed yet
+                CARDINALITY(stages.procnames) != 0
+                OR asmt.taxyr = DATE_FORMAT(NOW(), '%Y')
```

Comment:
Does this logic seem correct to you? It made sense in my QC, but it's hard to know for sure given that the only provisional values that currently exist in our database are pre-certified.